### PR TITLE
[docs] links to code style guides

### PR DESCRIPTION
**Pull request by Sjors**
I always have difficulty finding these, probably because they're not actually called "code style guideline".

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -186,7 +186,7 @@ In general, all pull requests must:
     the project (for example refactoring for modularisation);
   - Be well peer reviewed;
   - Have unit tests and functional tests where appropriate;
-  - Follow code style guidelines;
+  - Follow code style guidelines ([C++](doc/developer-notes.md), [function tests](test/functional/README.md));
```
Comment (MeshCollider) [inline CONTRIBUTING.md]: Nit: functional tests not function tests

**Comment by MeshCollider** [comment]
ACK https://github.com/bitcoin/bitcoin/pull/11729/commits/356a025d17a6ef3daad66ebc31494e4c8e15e49f

File: CONTRIBUTING.md (hunk 1 above)
Comment (Sjors) [reply CONTRIBUTING.md]: Fixed typo.

**Comment by practicalswift** [comment]
ACK fe27a72362ea8a165fd4648ee322541254d71228

Failed Travis job restarted!

**Comment by fanquake** [comment]
ACK fe27a72

**Comment by promag** [comment]
ACK fe27a72.

**Pull request merged**